### PR TITLE
Adding model serving with Hyperdisk and fixing READMEs

### PR DESCRIPTION
**Pull request by gushob21**

### CODE REVIEW

File: use-cases/inferencing/serving/vllm/autoscaling/README.md [hunk 1]
```diff
@@ -123,13 +122,13 @@ Service for Prometheus [documentation](https://cloud.google.com/kubernetes-engin
   - Queue-depth
 
     ```sh
-    kubectl --namespace ${SERVE_NAMESPACE} apply -f manifests/hpa-vllm-openai-queue-size.yaml
+    kubectl --namespace ${MLP_MODEL_OPS_NAMESPACE} apply -f manifests/hpa-vllm-openai-queue-size.yaml
```
Comment (arueth) [inline use-cases/inferencing/serving/vllm/autoscaling/README.md:125]: I think these should be `MLP_MODEL_SERVE_NAMESPACE` as that is where the model server is deployed

File: use-cases/inferencing/serving/vllm/autoscaling/README.md [hunk 2]
```diff
@@ -123,13 +122,13 @@ Service for Prometheus [documentation](https://cloud.google.com/kubernetes-engin
...
     ```
 
   - Batch-size
 
     ```sh
-    kubectl --namespace ${SERVE_NAMESPACE} apply -f manifests/hpa-vllm-openai-batch-size.yaml
+    kubectl --namespace ${MLP_MODEL_OPS_NAMESPACE} apply -f manifests/hpa-vllm-openai-batch-size.yaml
```
Comment (arueth) [inline use-cases/inferencing/serving/vllm/autoscaling/README.md:131]: Same as above

File: use-cases/inferencing/serving/vllm/autoscaling/README.md [hunk 3]
```diff
@@ -141,7 +140,7 @@ deployment pods when the metric goes over the specified threshold.
 - View the get the HPA status
 
   ```sh
-  kubectl --namespace ${SERVE_NAMESPACE} get/hpa vllm-openai-hpa --watch
+  kubectl --namespace ${MLP_MODEL_OPS_NAMESPACE} get hpa/vllm-openai-hpa --watch
```
Comment (arueth) [inline use-cases/inferencing/serving/vllm/autoscaling/README.md:143]: Same as above

File: use-cases/inferencing/serving/vllm/autoscaling/README.md [hunk 4]
```diff
@@ -152,7 +151,7 @@ deployment pods when the metric goes over the specified threshold.
 - As the metrics threshold is crossed, new pods would be created
 
   ```sh
-  kubectl --namespace ${SERVE_NAMESPACE} get pods --watch
+  kubectl --namespace ${MLP_MODEL_OPS_NAMESPACE} get pods --watch
```
Comment (arueth) [inline use-cases/inferencing/serving/vllm/autoscaling/README.md:154]: Same as above

File: use-cases/inferencing/serving/vllm/hyperdiskML/README.md [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+
+  ```sh
+  git clone https://github.com/GoogleCloudPlatform/accelerated-platforms && \
+  cd accelerated-platforms
+  ```
+
+- Change directory to the guide directory
+
+  ```sh
+  cd use-cases/inferencing/serving/vllm/hyperdiskML
```
Comment (arueth) [inline use-cases/inferencing/serving/vllm/hyperdiskML/README.md:28]: Update path based on comment above.

File: use-cases/inferencing/serving/vllm/hyperdiskML/README.md [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+## Prepare the HyperdiskML
+
+Loading model weights from a PersistentVolume is a method to load models faster. In GKE, PersistentVolumes backed by Google Cloud hyperdiskML can be mounted read-only simultaneously by multiple nodes (ReadOnlyMany), this allows multiple pods access to the model weights from a single volume.
+
+- Configure the environment
+
+  | Variable       | Description                                                                                  | Example                                |
+  | -------------- | -------------------------------------------------------------------------------------------- | -------------------------------------- |
+  | ACCELERATOR    | Type of GPU accelerator to use (l4, a100, h100)                                              | l4                                     |
+  | GCE_HYPERDISKML_NAME  | Name of the hyperdiskML that will host the model                                         | <unique_id>-vllm-model-weights-${ZONE} |
```
Comment (arueth) [inline use-cases/inferencing/serving/vllm/hyperdiskML/README.md:55]: Can you fix the spacing

**Review by arueth — CHANGES_REQUESTED** [review]
I didn't actually run through the README as it was similar enough to the persistent disk guide. I'll leave finally testing for the `int` branch before merging to `main`.

File: use-cases/inferencing/benchmark/README.md [hunk 7]
```diff
@@ -9,7 +9,7 @@ Refer to the documentation to [set up](https://docs.locust.io/en/stable/installa
 - A model is deployed using one of the vLLM guides
   - [Serving the mode using vLLM and GCSFuse](/use-cases/inferencing/serving/vllm/gcsfuse/README.md)
   - [Serving the mode using vLLM and Persistent Disk](/use-cases/inferencing/serving/vllm/persistent-disk/README.md)
-- Metrics are being scraped from the vLLM server ss shown in the [vLLM Metrics](/use-cases/inferencing/serving/vllm/metrics/README.md) guide.
```
[the remark this answers is not in the document]
Comment (gushob21) [reply use-cases/inferencing/benchmark/README.md:12]: yes, locust benchmarking in this guide don't require metrics. You can run benchmarking guide directly after deploying the model

File: use-cases/inferencing/serving/vllm/hyperdiskML/README.md (hunk 5 above)
Comment (gushob21) [reply use-cases/inferencing/serving/vllm/hyperdiskML/README.md:28]: The path `use-cases/inferencing/serving/vllm/hyperdiskML` seems to be correct to me, it is inline with other guides

File: use-cases/inferencing/serving/vllm/README.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+This guide uses `Single-Node Multi-GPU` method to serve the model that was fine-tuned in the previous guides.
+
+
+# Choosing storage to load model weights
+
+In order to serve a model on GKE, the model weights need to be download in the GKE container.
+There are different ways to load the model weights in the container:
+
+- Downlaod the model from Persisdent SSD disk - the model is loaded from a persistent disk.
```
Comment (kenthua) [inline use-cases/inferencing/serving/vllm/README.md:22]: typo

File: use-cases/inferencing/serving/vllm/README.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+- Download the model from GCS bucket - the model is loaded from a GCS bucket.
+- Download the model from Hyperdisk ML  - the model is loaded from high throughput HyperdiskML.
+- Use secondary boot disk - A disk can be preloaded with the model and be used as a secondary disk on the nodepools running the pods serving the model.
```
Comment (kenthua) [inline use-cases/inferencing/serving/vllm/README.md:25]: Focus on this as Container image loading, along with Image streaming?

File: use-cases/inferencing/serving/vllm/hyperdiskML/README.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+
+This guide demonstrates how to serve a model with vllm using hyperdiskML. By the end of this guide, you should be able to perform the following steps:
+
+- Create a hyperdiskML for the LLM model weights
+- Deploy a vLLM container to your cluster to host your model
+- Use vLLM to serve the fine-tuned Gemma model
+- View Production metrics for your model serving
+- Use custom metrics and Horizontal Pod Autoscaler (HPA) to scale your model
+
+## Prerequisites
```
Comment (kenthua) [inline use-cases/inferencing/serving/vllm/hyperdiskML/README.md:11]: a note here, or somewhere in the README that discusses performance benefits on larger models.

File: use-cases/inferencing/benchmark/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (arueth) [reply use-cases/inferencing/benchmark/README.md:12]: But wouldn't you ideally want to see the metrics for the model server as well?

File: use-cases/inferencing/serving/vllm/hyperdiskML/README.md [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+  ```
+
+  ```
+  job.batch/model-downloader condition met
+  complete
+  ```
+
+  Now, the model is downloaded to the persistent volume.
+
+### Create a HyperdiskML
```
Comment (kenthua) [inline use-cases/inferencing/serving/vllm/hyperdiskML/README.md:164]: product calls it `Hyperdisk ML`, so a global find/replace?

File: use-cases/inferencing/serving/vllm/hyperdiskML/README.md (hunk 5 above)
Comment (arueth) [reply use-cases/inferencing/serving/vllm/hyperdiskML/README.md:28]: This was in reference to the comment:

Can you change this to `hyperdiskml`, `hyperdisk-ml`, `hdml` for consistency of not using camel case.

**Review by kenthua — COMMENTED** [review]
global find/replace for `Hyperdisk ML`

File: use-cases/inferencing/benchmark/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (gushob21) [reply use-cases/inferencing/benchmark/README.md:12]: right, I stand corrected. Added it back

File: use-cases/inferencing/serving/vllm/README.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+# Serving the model
+
+In this guide, you will learn how to serve a model with vllm using the following storage options:
+
+- **Persistent Disk** : Follow the [guide](./persistent-disk/README.md) to serve a model with vllm using persistent disk.
+
+- **GCS** : Follow the [guide](./gcsfuse/README.md) to serve a model with vllm using GCSfuse download.
+
+- **HyperdiskML** : Follow the [guide](./hyperdiskML/README.md) to serve a model with vllm using HyperdiskML.
```
[the remark this answers is not in the document]
Comment (gushob21) [reply use-cases/inferencing/serving/vllm/README.md]: done

**Review by arueth — APPROVED** [review]

**Pull request merged**